Components/Translators/Sdk.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * Copyright Shopgate Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author    Shopgate Inc, 804 Congress Ave, Austin, Texas 78701 <interfaces@shopgate.com>
+ * @copyright Shopgate Inc
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ */
+
+namespace ShopgateCloudApi\Components\Translators;
+
+use Shopgate\CloudIntegrationSdk as ShopgateSdk;
+use Shopgate\CloudIntegrationSdk\ValueObject;
+use ShopgateCloudApi\Models\Auth\AccessToken;
+use ShopgateCloudApi\Models\Auth\RefreshToken;
+
+/**
+ * Helps translating Shopware objects to SDK
+ *
+ * @package ShopgateCloudApi\Components\Translators
+ */
+class Sdk
+{
+    protected static $headerTypes = ['Content-Type', 'Accept', 'Authorization'];
+
+    /**
+     * Retrieve SDK request using Shopware request
+     *
+     * @param \Enlight_Controller_Request_Request $request
+     *
+     * @return ValueObject\Request\Request
+     * @throws \Exception
+     */
+    public function getRequest(\Enlight_Controller_Request_Request $request)
+    {
+        $uri     = $request->getRequestUri();
```

Comment:
Instead of just passing the full uri, we should only pass the actual resource link, without any prefix (like "/frontend/shopgate" in this case).

Reply:
just to be clear, if the URI is: ```shopware/frontend/shopgate/v2/products/14```
The return should be ```v2/products/14```?

Reply:
now, that you bring it up, I just saw, that there is an inconsistency in the sdk code.
The path should start with `/v2`. But the sdk can't match `/v2/products/14`, yet. The `/v2` needs to be added to all the Route value objects and better be put into a V2 sub-directory (for later sdk versioning).